client/mock_test.go

```diff
@@ -40,23 +45,49 @@ func (m *MockClient) RoundAt(time time.Time) uint64 {
 }
 
 // ClientWithResults returns a client on which `Get` works `m-n` times.
-func MockClientWithResults(n, m int) Client {
+func MockClientWithResults(n, m uint64) *MockClient {
 	c := new(MockClient)
 	for i := n; i < m; i++ {
-		c.Results = append(c.Results, MockResult{uint64(i), []byte{byte(i)}})
+		c.Results = append(c.Results, NewMockResult(i))
 	}
 	return c
 }
 
+func NewMockResult(round uint64) MockResult {
+	sig := make([]byte, 8)
+	binary.LittleEndian.PutUint64(sig, round)
+	return MockResult{
+		rnd:  round,
+		sig:  sig,
+		rand: beacon.RandomnessFromSignature(sig),
+	}
+}
+
 type MockResult struct {
 	rnd  uint64
 	rand []byte
+	sig  []byte
+	_    struct{}
```

Comment:
why is this needed? the private fields mean others can't make this object alread, right?